### PR TITLE
WebRequest.DefaultWebProxy: Set doesn't work without a previous Get

### PR DESCRIPTION
**Pull request by arunjvs**
Iteration1: https://github.com/dotnet/corefx/pull/22720.
Added test case as per comments in iteration1.

Unless we do an unnecessary Get() on WebRequest.DefaultWebProxy, any Set() done will not be honored.

This is especially a problem on Linux, where DefaultWebProxy throws PlatformNotSupportedException(). This in turn breaks libraries like WindowsAzure.Storage which fail unless the Get() is called first and then WebRequest.DefaultWebProxy is set to null.

This fix avoids the unnecessary Get() by setting the already initialized flag sent to the LazyInitializer.

@davidsh @stephentoub

### CODE REVIEW

File: src/System.Net.Requests/tests/DummyWebProxy.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Test.Common
```
Comment (davidsh) [inline src/System.Net.Requests/tests/DummyWebProxy.cs]: This should not be the namespace for this class since this file is not in the Common\src directory.  It should be the same namespace as the other test files in this folder.

File: src/System.Net.Requests/tests/GlobalProxySelectionTest.cs [hunk 2]
```diff
@@ -3,47 +3,19 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
-
+using System.Net.Test.Common;
```
Comment (davidsh) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Remove this once you fix up the DummyWebProxy.cs file.

File: src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj [hunk 3]
```diff
@@ -8,10 +8,6 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Release|AnyCPU'" />
   <ItemGroup>
-    <Compile Include="HttpWebRequestTest.cs" />
-    <Compile Include="HttpWebResponseTest.cs" />
-    <Compile Include="RequestStreamTest.cs" />
-    <Compile Include="WebRequestTest.cs" />
     <Compile Include="$(CommonTestPath)\System\PlatformDetection.cs">
```
Comment (davidsh) [inline src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj]: Why are you moving these lines?  It isn't necessary as part of this change.

Comment (arunjvs) [inline src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj]: Organizing the csproj itemgroups. Let me know if it's not desirable.

Comment (davidsh) [inline src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj]: Please remove this change.  In general, you should keep your changes small and specific to the bug you are wishing to fix.

File: src/System.Net.Requests/tests/DummyWebProxy.cs (hunk 1 above)
Comment (arunjvs) [inline src/System.Net.Requests/tests/DummyWebProxy.cs]: Done

File: src/System.Net.Requests/tests/GlobalProxySelectionTest.cs (hunk 2 above)
Comment (arunjvs) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Done

File: src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj (hunk 3 above)
Comment (arunjvs) [inline src/System.Net.Requests/tests/System.Net.Requests.Tests.csproj]: Done

File: src/System.Net.Requests/tests/GlobalProxySelectionTest.cs [hunk 4]
```diff
@@ -3,47 +3,18 @@
...
-            }
-        }
-
         [Fact]
         public void Select_Success()
         {
             RemoteInvoke(() =>
             {
-                var myProxy = new MyWebProxy();
+                var myProxy = new DummyWebProxy();
```
Comment (stephentoub) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Does it need to be an instance of a new class like this?  Could it not just be a `new WebProxy()`?

Comment (arunjvs) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: WebProxy looks too generic a name for a dummy implementation. It sounds more like a base class for IWebProxy.

Comment (stephentoub) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: I'm not talking about a dummy implementation.  There's already a fairly generic WebProxy class that exists... I'm saying let's use it rather than adding a new file with a new dummy type... just use the type that already exists.

Comment (arunjvs) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Actually I realized it in indeed is. A separate dummy proxy still looks safe (keeping existing behavior in the tests). Let me know what you think.

Comment (stephentoub) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: I don't understand. Sure, it's safe. But it's unnecessary code. Why an unnecessary code to the repo?

Comment (arunjvs) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Okay so should I keep GlobalProxySelectionTest as is with it's MyWebProxy or change it to WebProxy along with WebRequestTest?

Comment (davidsh) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Revert the change to GlobalProxySelectionTest.

Then try the simple "WebProxy()" change for your new test.

Comment (arunjvs) [inline src/System.Net.Requests/tests/GlobalProxySelectionTest.cs]: Done! Works with WebProxy().

**Comment by arunjvs** [comment]
Addressed comments and validated the test works with the change.
